File: java/src/main/java/mx/iteso/adapter/word/WordDocumentAdapter.java
package mx.iteso.adapter.word;

import mx.iteso.adapter.Font;
import mx.iteso.adapter.Image;
import mx.iteso.adapter.Style;
import mx.iteso.adapter.googledoc.GoogleDoc;

/**
 * WordDocumentAdapter Class.
 */
public class WordDocumentAdapter extends GoogleDoc {
    /**
     * Attr type WordDocument.
     */
    private WordDocument wordDocument;
    /**
     * Public constructor.
     * @param newWordDocument WordDocument for the adapter.
     */
    public WordDocumentAdapter(final WordDocument newWordDocument) {
        super();
        this.wordDocument = newWordDocument;
        setSharingPermissions((int) wordDocument.getLicense()
                                                .getLicenseVersion());
    }
    /**
     * Font getter.
     * @return Font of the WordDocument.
     */
    @Override
    public Font getFont() {
        if (wordDocument.getFormat() != null) {
            return wordDocument.getFormat().getFont();
        }
        return null;
    }
    /**
     * Backgroung image getter.
     * @return WordDocument's background image.
     */
    @Override
    public Image getBackground() {
        return wordDocument.getBackground();
    }
    /**
     * Style getter.
     * @return WordDocument's Style.
     */
    @Override
    public Style getStyle() {
        if (wordDocument.getFormat() != null) {
            return new Style(wordDocument.getFormat().getFont());
        }
        return null;
    }
    /**
     * Sharing permissions getter.
     * @return WordDocument's sharing permissions.
     */
    @Override
    public int getSharingPermissions() {
        return super.getSharingPermissions();
    }
}

File: python/src/main/python/WordDocumentAdapter.py
package mx.iteso.adapter;

/**Word Document Adapter. */
public class WordDocumentAdapter extends GoogleDoc {
    /**Word document. */
    private WordDocument wordDocument;
    /**Background image. */
    private BackgroundImage img;
    /**Share permissions. */
    private int sharePerms;
    /**Edit permissions. */
    private static final int EDIT_PERMISSIONS = 3;
    /**Valid version. */
    private static final float VALID_VERSION = 16.0F;
    /**see permissions. */
    private static final int SEE_PERMISSIONS = 1;
    /**Constructor.
     * @param nWordDocument Word Document.
     */
    public WordDocumentAdapter(final WordDocument nWordDocument) {
        this.wordDocument = nWordDocument;
        this.img = new BackgroundImage(
            this.wordDocument.getBackground().getPath(),
            (int) this.wordDocument.getBackground().getWidth(),
            (int) this.wordDocument .getBackground().getHeight());
        this.sharePerms = (this.wordDocument.getLicense()
        .getVersion() > VALID_VERSION) ? EDIT_PERMISSIONS : SEE_PERMISSIONS;
    }
    /**
     * @return background.
     */
    @Override
    public BackgroundImage getBackground() {
        return this.img;
    }
    /**
     * @return Sharing permissions.
     */
    @Override
    public int getSharingPermissions() {
        return this.sharePerms;
    }
}
